### PR TITLE
Add environment variable when instantiating clients

### PR DESCRIPTION
**Pull request by ddl-gabrielhaim**
https://dominodatalab.atlassian.net/browse/DOM-48748

### CODE REVIEW

File: R/client.R [hunk 1]
```diff
@@ -23,8 +23,17 @@
...
+      domino_data_sources$DataSourceClient(api_key, token_file)
+    )
+  } else {
+    client <- withr::with_envvar(
+      new = envvar,
+      domino_data_sources$DataSourceClient()
+    )
   }
-  domino_data_sources$DataSourceClient()
+  return(client)
```
Comment (ronit-basu) [inline R/client.R:38]: TIL scala does scope interestingly

File: R/client.R [hunk 2]
```diff
@@ -23,8 +23,17 @@
 #' @return A `domino_data.data_sources.DataSourceClient`.
 #' @export
 datasource_client <- function(api_key = NULL, token_file = NULL) {
+  envvar <- c("DOMINO_CLIENT_SOURCE" = "R")
   if (!is.null(api_key) || !is.null(token_file)) {
-    return(domino_data_sources$DataSourceClient(api_key, token_file))
+    client <- withr::with_envvar(
```
Comment (ronit-basu) [inline R/client.R:28]: nice!

**Review by ronit-basu — APPROVED** [review]

File: R/client.R (hunk 1 above)
Comment (ddl-gabrielhaim) [reply R/client.R:38]: scala?

Comment (ddl-mcetin) [reply R/client.R:38]: i am confused too

**Review by ddl-giuliocapolino — APPROVED** [review]

**Pull request merged**